### PR TITLE
Add remaning sklearn metrics

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

This PR adds remaining metrics for sklearn backend

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/metrics/sklearns.py [hunk 1]
```diff
@@ -232,6 +236,57 @@ def forward(
...
+                Defaults to all processes (world)
+            reduce_op: the operation to perform during reduction within DDP (only needed for DDP training).
+                Defaults to sum.
+        """
+        super().__init__('balanced_accuracy_score',
+                         reduce_group=reduce_group,
+                         reduce_op=reduce_op,
+                         adjusted=adjusted)
+
+    def forward(self, y_true: np.ndarray, y_pred: np.ndarray,
```
Comment (justusschock) [inline pytorch_lightning/metrics/sklearns.py:274]: @SkafteNicki can we switch y_true and y_pred here? I've done that to be consistent with PyTorch notation where its first red and then target.

Comment (SkafteNicki) [reply pytorch_lightning/metrics/sklearns.py:274]: Of cause :)

**Comment by Borda** [comment]
@SkafteNicki can we finish it for 0.9?

**Comment by SkafteNicki** [comment]
@Borda yes, it should be ready to go as soon as I fix the failing tests

**Comment by SkafteNicki** [comment]
@Borda and @justusschock this should be ready to merge now

File: tests/metrics/test_sklearn.py [hunk 2]
```diff
@@ -16,21 +16,51 @@
...
+    AUROC,
+    MeanAbsoluteError,
+    MeanSquaredError,
+    MeanSquaredLogError,
+    MedianAbsoluteError,
+    R2Score,
+    MeanPoissonDeviance,
+    MeanGammaDeviance,
+    MeanTweedieDeviance,
+    ExplainedVariance
```
Comment (Borda) [inline tests/metrics/test_sklearn.py:63]:
```suggestion
    ExplainedVariance,
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -29,6 +29,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Added support for Mean in DDP Sync ([#2568](https://github.com/PyTorchLightning/pytorch-lightning/pull/2568))
 
+- Added remaining sklearn metrics ([#2562](https://github.com/PyTorchLightning/pytorch-lightning/pull/2562))
```
Comment (Borda) [inline CHANGELOG.md:32]:
```suggestion
- Added remaining `sklearn` metrics: `AveragePrecision`, `BalancedAccuracy`, `CohenKappaScore`, `DCG`, `Hamming`, `Hinge`, `Jaccard`, `MeanAbsoluteError`, `MeanSquaredError`, `MeanSquaredLogError`, `MedianAbsoluteError`, `R2Score`, `MeanPoissonDeviance`, `MeanGammaDeviance`, `MeanTweedieDeviance`, `ExplainedVariance` ([#2562](https://github.com/PyTorchLightning/pytorch-lightning/pull/2562))
```

**Comment by Borda** [comment]
just fixed RMSE and ready to go 🎉

**Pull request merged**